src/cgame/rocket/rocketProgressBar.h

```diff
@@ -36,289 +36,50 @@ Maryland 20850 USA.
 
-enum progressBarOrientation_t
-{
-	LEFT,
-	RIGHT,
-	UP,
-	DOWN
-};
-
-class RocketProgressBar : public Rml::Core::Element
+class RocketProgressBar : public Rml::Controls::ElementProgressBar
 {
 public:
 	RocketProgressBar( const Rml::Core::String &tag ) :
-		Rml::Core::Element( tag ), orientation( LEFT ), value( 0.0f ), shader( 0 ), color( Color::White )
-	{
-	}
-
-	~RocketProgressBar() {}
-
-	// Get current value
-	float GetValue() const
+		Rml::Controls::ElementProgressBar( tag ), value( 0.0f ), source( "" )
 	{
-		return value;
 	}
 
-	void SetValue( const float value )
-	{
-		SetAttribute( "value", value );
-	}
-
-
-	void Update()
-	{
-		float newValue;
-
-		if ( !source.empty() )
-		{
-			newValue = CG_Rocket_ProgressBarValue(source.c_str());
-
-			if ( newValue != value )
-			{
-				SetValue( newValue );
-			}
-		}
-	}
-
-	void OnRender()
-	{
-
-		if ( !shader )
-		{
-			return;
-		}
-
-		Update();
-		Rml::Core::Vector2f position = GetAbsoluteOffset();
-
-		// Vertical meter
-		if( orientation >= UP )
-		{
-			float height;
-			height = dimensions.y * value;
-
-			// Meter decreases down
-			if ( orientation == DOWN )
-			{
-				trap_R_SetColor( color );
-				trap_R_DrawStretchPic( position.x, position.y, dimensions.x, height,
-						       0.0f, 0.0f, 1.0f, value, shader );
-				trap_R_ClearColor();
-			}
-			else
-			{
-				trap_R_SetColor( color );
-				trap_R_DrawStretchPic( position.x, position.y - height + dimensions.y, dimensions.x,
-						       height, 0.0f, 1.0f - value, 1.0f, 1.0f, shader );
-				trap_R_ClearColor();
-			}
-		}
-
-		// Horizontal meter
-		else
-		{
-			float width;
-
-			width = dimensions.x * value;
-
-			// Meter decreases to the left
-			if ( orientation == LEFT )
-			{
-				trap_R_SetColor( color );
-				trap_R_DrawStretchPic( position.x, position.y, width,
-						       dimensions.y, 0.0f, 0.0f, value, 1.0f, shader );
-				trap_R_ClearColor();
-			}
-			else
-			{
-				trap_R_SetColor( color );
-				trap_R_DrawStretchPic( position.x - width + dimensions.x, position.y, width,
-						       dimensions.y, 1.0f - value, 0.0f, 1.0f, 1.0f, shader );
-				trap_R_ClearColor();
-			}
-		}
-	}
-
-	void OnPropertyChange( const Rml::Core::PropertyIdSet &changed_properties )
-	{
-		Element::OnPropertyChange( changed_properties );
-
-		if ( changed_properties.Contains( Rml::Core::PropertyId::Color ) )
-		{
-			color = Color::Adapt( GetProperty( Rml::Core::PropertyId::Color )->Get<Rml::Core::Colourb>() );
-		}
-
-		if ( changed_properties.Contains( UnvPropertyId::Orientation ) )
-		{
-			Rml::Core::String  orientation_string = GetProperty( UnvPropertyId::Orientation )->Get<Rml::Core::String>();
-
-			if ( orientation_string == "left" )
-			{
-				orientation = LEFT;
-			}
-			else if ( orientation_string == "up" )
-			{
-				orientation = UP;
-			}
-			else if ( orientation_string == "down" )
-			{
-				orientation = DOWN;
-			}
-			else
-			{
-				orientation = RIGHT;
-			}
-		}
-	}
+	virtual ~RocketProgressBar() {}
```

Comment:
override?
Also, there is the ` = default` thing. It's a bit longer, though, but makes it easy to notice with syntax coloration. Dunno about it's use.

Reply:
I meant, since the class inherits, the DTor could use override instead.